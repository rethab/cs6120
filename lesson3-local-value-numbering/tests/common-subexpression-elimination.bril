@main {
  a: int = const 1;
  b: int = const 2;
  c: int = add a b;
  d: int = add a b;
  e: int = mul c d;
  print e;
}
