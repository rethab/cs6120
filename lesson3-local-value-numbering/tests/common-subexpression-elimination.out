@main {
  e: int = const 9;
  print e;
}
